### DISCUSSION TITLE
How to Bundle Multiple Architectures for WinUI App Submission to Microsoft Store?

### DISCUSSION THREAD

**Discussion by nextcodelab** [microsoft/microsoft-ui-xaml#9450] [Q&A] [answered]
Hello WinUI community,

I'm currently working on developing a WinUI (Windows UI Library) application and planning to submit it to the Microsoft Store. However, unlike UWP (Universal Windows Platform) apps, where bundling for multiple architectures is automatic, I'm facing challenges in bundling my WinUI app for multiple architectures.

I have developed my WinUI app to support x86, x64, and ARM64 architectures, targeting a broad range of Windows devices. However, I'm unsure about the correct approach to bundle these different architecture packages together for submission to the Microsoft Store.

Could someone please provide guidance or best practices on how to bundle WinUI app packages for multiple architectures manually? Additionally, any insights or tips on optimizing the bundling process and ensuring compatibility across different Windows device types would be highly appreciated.

The current output file in create packages in winui3 is something like this [Image Below ], unlike uwp is automatically bundled.
![Screenshot 2024-03-16 101256](https://github.com/microsoft/microsoft-ui-xaml/assets/73583588/a02a0d37-e064-4250-b95d-de30635d0700)

[https://github.com/microsoft/devhome](https://github.com/microsoft/devhome)
Windows.DevHome repo  bundled the app like this.

[https://github.com/microsoft/devhome/releases/download/v0.1101.416.0/Windows.DevHome_0.1101.416.0.msixbundle](https://github.com/microsoft/devhome/releases/download/v0.1101.416.0/Windows.DevHome_0.1101.416.0.msixbundle)

Thank you in advance for your help and insights!

**Comment by DarranRowe** [comment, marked as answer]
This is currently a known issue and one draw back of the WinUI 3 with the packaging in the same project.
The project type with the separate application packaging project will bundle properly. Unfortunately, you would need to use makeappx directly to bundle the separate .msix files into a .msixbundle.
It is pretty easy to bundle.
1) Copy the .msix files that you want to bundle into a separate directory. The contents of that directory will be the contents of the bundle.
2) Use the command `makeappx bundle /d [input_directory] /p [bundle_path].msixbundle`

**Comment by nextcodelab** [comment]
Thank you so much....
# 1. makeappx.exe
   C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64\makeappx.exe
# 2. msix files
   C:\Users\Robel\Documents\AppBundle\Bundles\Msix
# 3. Output .msixbundle file
   C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle

# CMD
```cs
& "C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64\makeappx.exe" bundle /d "C:\Users\Robel\Documents\AppBundle\Bundles\Msix" /p "C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle"

```

# Final
```cs
PS C:\Users\Robel\Desktop> & "C:\Program Files (x86)\Windows Kits\10\bin\10.0.22621.0\x64\makeappx.exe" bundle /d "C:\Users\Robel\Documents\AppBundle\Bundles\Msix" /p "C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle"
Microsoft (R) MakeAppx Tool
Copyright (C) 2013 Microsoft.  All rights reserved.

The path (/p) parameter is: "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle"
The content directory (/d) parameter is: "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\Msix"
Enumerating files from directory "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\Msix"
Packing 1 file(s) in "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\Msix" (content directory) to "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle" (output file name).
File "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\MyApp.msixbundle" already exists.  Overwrite? [Y]es (any other key to skip file): y
Adding "\\?\C:\Users\Robel\Documents\AppBundle\Bundles\Msix\WinUIGallery_1.1.82.0_arm64.msix" to the bundle as a payload package.  Its path in the bundle is "WinUIGallery_1.1.82.0_arm64.msix".
Bundle creation succeeded.
PS C:\Users\Robel\Desktop>
```

# Please do not delete this for future reference...
